mathics/core/expression.py

```diff
@@ -594,6 +594,10 @@ def round_to_float(self, evaluation=None, permit_complex=False):
         if evaluation is None:
             value = self
         elif isinstance(evaluation, sympy.core.numbers.NaN):
+            print(
+                "why an evaluation parameter would be set to sympy.NaN and this should happend? "
+            )
+            assert False
```

Comment:
This is essentially the same as raise `RuntimeError("your message here")1

Reply:
Indeed, but I also wanted to know why this check is done...